### PR TITLE
bpo-40455: Remove gcc10 warning about x_digits

### PR DESCRIPTION
**Pull request by corona10**
https://bugs.python.org/issue40455

### CODE REVIEW

File: Objects/longobject.c [hunk 1]
```diff
@@ -2903,8 +2904,6 @@ _PyLong_Frexp(PyLongObject *a, Py_ssize_t *e)
         shift_digits = (DBL_MANT_DIG + 2 - a_bits) / PyLong_SHIFT;
         shift_bits = (DBL_MANT_DIG + 2 - a_bits) % PyLong_SHIFT;
         x_size = 0;
```
Comment (mdickinson) [inline Objects/longobject.c]: I think you want `x_size = shift_digits` here. Note that the block being removed changes the value of `x_size`, and that `x_size` is used after the block.

**Review by mdickinson — CHANGES_REQUESTED** [review]
The approach looks good to me, but there's a logic error to fix. I'm a little bit disturbed that we didn't get more test failures here.

File: Objects/longobject.c (hunk 1 above)
Comment (corona10) [reply Objects/longobject.c]: Thank you for catching this!

**Comment by corona10** [comment]
Thank you for the review and for catching the missed thing :)
I have made the requested changes; please review again

**Review by mdickinson — APPROVED** [review]
LGTM! Thank you.

**Comment by corona10** [comment]
@vstinner Since I am still in mentorig period, I will merge it after your confirm :)

**Review by shihai1991 — APPROVED** [review]
LGTM.

**Review by vstinner — APPROVED** [review]
LGTM.

My main worry is that this change will initialize twice a few digits and so it would make the code less efficient. But @mdickinson approved the change, so I'm fine with it :-)

The change is correct. Initializing digits to zero cannot be wrong :-)

**Pull request merged**